test/test_string.cpp

```diff
@@ -106,9 +106,21 @@ TORRENT_TEST(string_equal_no_case)
 	TEST_CHECK(string_equal_no_case("foobar", "foobar"));
 	TEST_CHECK(!string_equal_no_case("foobar", "foobar "));
 	TEST_CHECK(!string_equal_no_case("foobar", "F00"));
+	TEST_CHECK(!string_equal_no_case("foobar", "foo"));
+	TEST_CHECK(!string_equal_no_case("foo", "foobar"));
 
 	TEST_CHECK(string_begins_no_case("foobar", "FoobAR --"));
 	TEST_CHECK(!string_begins_no_case("foobar", "F00"));
+	TEST_CHECK(!string_begins_no_case("foobar", "foo"));
+	TEST_CHECK(string_begins_no_case("foo", "foobar"));
```

Comment:
I think it would be clearer if you would group the positive test cases together, and the negative test cases together